testing/trino-faulttolerant-tests/pom.xml

```diff
@@ -153,13 +153,83 @@
             <groupId>io.trino</groupId>
             <artifactId>trino-delta-lake</artifactId>
             <scope>test</scope>
+            <exclusions>
+                <!-- Avoid dependency conflicts with the Google libraries pulled by trino-exchange-filesystem -->
```

Comment:
> I'd prefer to see no exclusions here and instead, use a consistent version. https://github.com/trinodb/trino/pull/12523/files#r962920057 could help.

This is rather hard to achieve at the time of this writing due to the fact that bigquery connector, google sheets connector , delta lake connector each use its own version of google dependencies.


Note that, at the time of this writing, trino-faulttolerant-tests project uses delta-lake to test against S3 minio implementation.

I remember we've talked offline about this topic and we've agreed that as long as the version conflicts between the google libraries is limited to the `trino-faulttolerant-tests` we can consider this conflict as acceptable.
cc @alexjo2144

Reply:
> I remember we've talked offline about this topic and we've agreed that as long as the version conflicts between the google libraries is limited to the trino-faulttolerant-tests we can consider this conflict as acceptable.

Acceptable... yeah.

> Note that, at the time of this writing, trino-faulttolerant-tests project uses delta-lake to test against S3 minio implementation.

Good point. @losipiuk any plans to test faulttolerant against GCS as well?

> This is rather hard to achieve at the time of this writing due to the fact that bigquery connector, google sheets connector , delta lake connector each use its own version of google dependencies.

that's not good.
that means
- a potential CVE needs to be fixed multiple times; easy to miss a place that requires an update
- bigger installation - no jar deduplication since versions differ

do we have an issue to fix that?

Reply:
> do we have an issue to fix that?

Bring google dependency versions in sync within the projects: 

- `trino-hive` (& `trino-delta-lake` and probably `trino-iceberg`)
- `trino-google-sheets`
- `trino-exchange-filesystem`
- `trino-bigquery`


Is this what you mean? 

I haven't found any corresponding existing issue.


cc: @ebyhr

Reply:
let's create one

Reply:
> Good point. @losipiuk any plans to test faulttolerant against GCS as well?

No explicit plans I think. It would be great to have those but currently we are not invesing in any tests on "real" cloud providers. We are only testing against minio, believing it is good enough simulation of AWS S3. 
It may backfire I think. @linzebing @arhimondr what do you think about adding a minimalistic "product" tests for filesystem exchange when we use that agains real GCE, AWS, Azure?

Reply:
> let's create one

@findepi  I'm thinking that you imply creating a GCS test within `trino-faulttolerant-tests` project.
With `trino-exchange-filesystem` and `trino-delta-lake` in the same class loader it is quite likely that sooner or later we'll hit into a method missing/  class loader related issue. This is why I am reluctant to adding a GCS based test within `trino-faulttolerant-tests` project.

I added however a basic product test for the write operations on GCS.

Reply:
Currently we may not spend time on adding tests against GCS (it basically shares the same code path as AWS S3).